File: Alay/Sliding-Window/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.size() == 0) return 0;
        vector<int> arr(256, 0);
        int start = 0, end = 0, maxLen = 0;
        while(end < s.size()){
            int d = s[end];
            arr[d]++;
            while(arr[d] > 1){
                int di = s[start];
                arr[di]--;
                start++;
            }
            maxLen = max(maxLen, end - start + 1);
            end++;
        }
        return maxLen;
    }
};

File: Aditya-Pratap/Sliding Window/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) <= 1:
            return len(s)
        
        arr = [0] * 128
        a, e = 0, 0
        maxLength = 0
        
        while e < len(s):
            diff = s[e]
            arr[ord(diff)] += 1
            
            while arr[ord(diff)] > 1:
                d = s[a]
                arr[ord(d)] -= 1
                a += 1
            
            maxLength = max(maxLength, e - a + 1)
            e += 1
        
        return maxLength